packages/playground/src/portal/transfer_view.vue

```diff
@@ -0,0 +1,221 @@
+                @click="submitFormAddress"
+                >Send</v-btn
+              >
+            </v-card-actions>
+          </v-card>
+        </v-window-item>
+      </v-window>
+    </v-card>
+  </v-container>
+</template>
+<script lang="ts" setup>
+import { Keyring } from "@polkadot/keyring";
+import { Client, QueryClient, type Twin } from "@threefold/tfchain_client";
+import Decimal from "decimal.js";
+import { createToast } from "mosha-vue-toastify";
+import { onMounted, ref } from "vue";
+
+import { useProfileManager } from "../stores";
+import { getGrid, loadBalance } from "../utils/grid";
+
+onMounted(async () => {
+  await getFreeBalance();
+});
+
+const freeBalance = ref(0);
+const activeTab = ref(0);
+const receipientTwinId = ref("");
+const isValidTwinIDTransfer = ref(false);
+const transferAmount = ref(1);
+const loadingTwinIDTransfer = ref(false);
+const loadingAddressTransfer = ref(false);
+const isValidAddressTransfer = ref(false);
+const receipientAddress = ref("");
+const profile = useProfileManager().profile;
+
+const queryClient = new QueryClient(window.env.SUBSTRATE_URL);
+const client = new Client({
+  url: window.env.SUBSTRATE_URL,
+  mnemonicOrSecret: useProfileManager().profile!.mnemonic,
+});
+function isSameAddress() {
+  if (receipientAddress.value == profile?.address) {
+    return false;
+  }
+  return true;
+}
+function isValidAddress() {
+  const keyring = new Keyring({ type: "sr25519" });
+  try {
+    keyring.addFromAddress(receipientAddress.value);
+
+    return true;
+  } catch (error) {
+    return false;
+  }
+}
+
+function clearInput() {
+  transferAmount.value = 1;
+  receipientTwinId.value = "";
+  receipientAddress.value = "";
+}
+async function getFreeBalance() {
+  const grid = await getGrid(profile!);
+  const balance = await loadBalance(grid!);
+  freeBalance.value = balance.free;
+}
+async function transfer(receipientTwin: Twin) {
+  const twinAddress = receipientTwin.accountId;
+
+  const decimalAmount = new Decimal(transferAmount.value);
+  const milliAmount = decimalAmount.mul(10 ** 7).toNumber();
+  try {
+    const transferResult = await client.balances.transfer({
+      address: twinAddress,
+      amount: milliAmount,
+    });
+    await transferResult.apply();
+    createToast("Transaction submitted!", {
+      position: "top-right",
+      hideProgressBar: true,
+      toastBackgroundColor: "green",
+      timeout: 5000,
+    });
+
+    await getFreeBalance();
+  } catch (err) {
+    createInvalidTransferToast("transfer failed!");
+  }
+}
+async function submitFormAddress() {
+  const twinId = await queryClient.twins.getTwinIdByAccountId({ accountId: receipientAddress.value });
+  const twinDetails = await queryClient.twins.get({ id: twinId });
+  if (twinDetails != null) {
+    loadingAddressTransfer.value = true;
+    await transfer(twinDetails);
+    loadingAddressTransfer.value = false;
+  } else {
+    createInvalidTransferToast("twin ID doesn't exist");
```

Comment:
can't we validate the input and show the error in place instead of showing toast?